universe/common/unity/options_auto.go

```diff
@@ -0,0 +1,190 @@
+package unity
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/momentum-xyz/posbus-protocol/posbus"
+	"github.com/momentum-xyz/ubercontroller/config"
+	"github.com/momentum-xyz/ubercontroller/types/entry"
+	"github.com/momentum-xyz/ubercontroller/universe"
+	"github.com/momentum-xyz/ubercontroller/universe/common/api/dto"
+	"github.com/momentum-xyz/ubercontroller/utils"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+)
+
+type AttributeValueChangedMessage struct {
+	Type universe.AttributeChangeType     `json:"type"`
+	Data AttributeValueChangedMessageData `json:"data"`
+}
+
+type AttributeValueChangedMessageData struct {
+	AttributeName string `json:"attribute_name"`
+	Value         any    `json:"value"`
+}
+
+type AttributeSlotKind string
+
+const (
+	AttributeSlotKindInvalid AttributeSlotKind = ""
+	AttributeSlotKindTexture AttributeSlotKind = "texture"
+	AttributeSlotKindVideo   AttributeSlotKind = "video"
+	AttributeSlotKindNumber  AttributeSlotKind = "number"
+	AttributeSlotKindString  AttributeSlotKind = "string"
+	AttributeSlotKindImage   AttributeSlotKind = "image"
+)
+
+func GetOptionAutoOption(options *entry.AttributeOptions) (*entry.UnityAutoAttributeOption, error) {
+	if options == nil {
+		return nil, nil
+	}
+
+	autoOptionsValue, ok := (*options)["unity_auto"]
+	if !ok {
+		return nil, nil
+	}
+
+	var autoOption *entry.UnityAutoAttributeOption
+	if err := utils.MapDecode(autoOptionsValue, autoOption); err != nil {
+		return nil, errors.WithMessage(err, "failed to decode auto option")
+	}
+
+	return autoOption, nil
+}
+
+func GetOptionAutoMessage(option *entry.UnityAutoAttributeOption, changeType universe.AttributeChangeType,
+	attributeID entry.AttributeID, value *entry.AttributeValue) (*websocket.PreparedMessage, error) {
+
+	// do checks if obligatory fields are present
+	if option == nil {
+		return nil, nil
+	}
+	if option.SlotKind == "" || option.ContentType == "" {
+		return nil, nil
+	}
+
+	data := &AttributeValueChangedMessage{
+		Type: changeType,
+		Data: AttributeValueChangedMessageData{
+			AttributeName: attributeID.Name,
+			Value:         value,
+		},
+	}
+
+	// do some checks depending on slot kind
+	// for textures we need to render them depending on the
+	// content type: currently there are 2 options:
+	// video and text/number/string
+	switch option.SlotKind {
+	case "texture":
+		if option.ContentType == "video" {
+			payload, err := json.Marshal(map[string]interface{}{
```

Comment:
actualy we use "any" instead of "interface" for things like this, to avoid confusing with interfaces